packages/runtime/src/interpreter/opcode-list.ts

```diff
@@ -4073,24 +4066,35 @@ export class ITxnSubmit extends Op {
 		this.interpreter = interpreter;
 	}
 
+	// eslint-disable-next-line sonarjs/cognitive-complexity
 	execute(_stack: TEALStack): void {
 		if (typeof this.interpreter.subTxn === "undefined") {
 			throw new RuntimeError(RUNTIME_ERRORS.TEAL.ITXN_SUBMIT_WITHOUT_ITXN_BEGIN, {
 				line: this.line,
 			});
 		}
 
+		// back up current context
+		if (this.interpreter.runtime.remainCtx === undefined) {
+			this.interpreter.runtime.remainCtx = cloneDeep(this.interpreter.runtime.ctx);
+		}
+
 		// calculate fee accross all txns
 		let totalFee = 0;
 		for (const t of this.interpreter.runtime.ctx.gtxs) {
 			totalFee += t.fee ?? 0;
 		}
+
 		for (const t of this.interpreter.innerTxns) {
 			totalFee += t.fee ?? 0;
 		}
+
 		totalFee += this.interpreter.subTxn.fee ?? 0;
-		const totalTxCnt =
-			this.interpreter.runtime.ctx.gtxs.length + this.interpreter.innerTxns.length + 1;
+
+		let totalTxCnt = this.interpreter.innerTxns.length + 1;
+		if (!this.interpreter.runtime.ctx.isInnerTx) {
+			totalTxCnt += this.interpreter.runtime.ctx.gtxs.length;
```

Comment:
what if other transactions in the group have inner transactions ?

Reply:
Added task for this case.